src/nvim/edit.c

```diff
@@ -4239,9 +4239,10 @@ void ins_compl_check_keys(int frequency)
 static int ins_compl_key2dir(int c)
 {
   if (c == Ctrl_P || c == Ctrl_L
-      || (pum_visible() && (c == K_PAGEUP || c == K_KPAGEUP
-                            || c == K_S_UP || c == K_UP)))
+      || c == K_PAGEUP || c == K_KPAGEUP
+      || c == K_S_UP || c == K_UP) {
```

Comment:
Because, it is not `pum_visible()` state in 2nd `ins_complete()`.
Please see below `Lazily show the popup menu, unless we got interrupted.`.